### ISSUE TITLE
smtp_forward times out in 2.8.7, but not in 2.7.3

### ISSUE THREAD

**Issue by randunel** [haraka/Haraka#1533]
### Haraka version

2.8.7 => time out

2.7.3 => forwarded successfully
### Expected behavior

2.8.7 => forwarded successfully
### Observed behavior

When using Haraka 2.8.7 and the same configs, files and plugins I use with 2.7.3, smtp_forward times out.

I also observe the following log line with every forward attempt `[ERROR] [-] [core] client.socket.remoteAddress undefined!`. I don't know if it's related or not.
### Steps to reproduce

Use the following plugins:
- rcpt_to.in_host_list
- queue/smtp_forward

Use the following `config/smtp_forward.ini` values:

```
host=smtp.sendgrid.net
port=587
enable_tls=1
auth_type=plain
auth_user=...
auth_password=...
```

Try to forward any e-mail via sendgrid, it will _not_ work using haraka `2.8.7`, but it _does_ work using `2.7.3`.

**Comment by Dexus** [comment]
Wich version of nodejs?

It's a known bug of NodeJs that there is not always set the remote-address in some versions.

Von meinem iPhone gesendet

**Comment by msimerson** [comment]
I'd just like to point out that this error:

> [ERROR] [-] [core] client.socket.remoteAddress undefined!

Is exactly the error that will be returned if the smtp_forward attempts to connect to the remote server but that remote server isn't available.

**Comment by ajay-psd** [comment]
Hi

I also tried smtp_forward plugin, but it did not worked, i am using haraka 2.8.9 version
i have used two plugin
**auth/flat_file**                     ; for authenticating and receiving mails
**queue/smtp_forward**       ; forwarding to sendgrid

config/smtp_forward.ini
host=smtp.sendgrid.net
port=587
enable_tls=1
auth_type=plain
auth_user=...
auth_password=...

**I have tried telnet and connect smtp.sendgrid.net from same server, it is working, means remote server is available.**

I was assuming authentication is having a problem. but I verified auth_user and password is  correct. and also i tried with wrong credentials its giving same result.
  
I am not able to understand where the issue is? Kindly suggest on this!

Below is haraka log   
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] S: 250 recipient <ajay@gmail.com> OK
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] C: DATA state=1
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] running data hooks
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] S: 354 go ahead, make my day
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] running data_post hooks
[NOTICE] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] message mid="<f2074a24ff99f1f4ea528b27dcc247cd@localhost.localdomain>" size=1699 rcpts=1/0/0 delay=0.001 code=CONT msg=""
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] running queue_outbound hooks
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] running queue_outbound hook in queue/smtp_forward plugin
[INFO] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] forwarding to smtp.sendgrid.net:587
[INFO] [-] [core] [smtp_client_pool] [587:smtp.sendgrid.net:300] dispense() clients=1 available=0
[DEBUG] [-] [core] [smtp_client_pool] [587:smtp.sendgrid.net:300] createResource() - creating obj - count=1 min=0 max=1000
[DEBUG] [-] [core] [smtp_client_pool] uuid=502FB3D1-C7D6-4938-9D76-64DFE8E2F0FD host=smtp.sendgrid.net port=587 pool_timeout=300 created
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] Got smtp_client: 502FB3D1-C7D6-4938-9D76-64DFE8E2F0FD
[INFO] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] Configuring authentication for SMTP server smtp.sendgrid.net:587
**[ERROR] [-] [core] client.socket.remoteAddress undefined!**
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 220 SG ESMTP service ready at ismtpd0019p1sin1.sendgrid.net\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] C: EHLO haraka.test
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250-smtp.sendgrid.net\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250-8BITMIME\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250-PIPELINING\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250-SIZE 31457280\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250-STARTTLS\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250-AUTH PLAIN LOGIN\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 250 AUTH=PLAIN LOGIN\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] C: STARTTLS
[INFO] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] Authenticating with AUTH PLAIN userajay
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] C: AUTH PLAIN AHVzZXJhamF5AHVuZGVmaW5lZA==
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] S: 220 Begin TLS negotiation now\r\n
[PROTOCOL] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [queue/smtp_forward] C: MAIL FROM:<noreply@sinfinity.com>
[DEBUG] [-] [core] [smtp_client_pool] 502FB3D1-C7D6-4938-9D76-64DFE8E2F0FD destroyed, state=2
[DEBUG] [-] [core] [smtp_client_pool] 502FB3D1-C7D6-4938-9D76-64DFE8E2F0FD: [smtp.sendgrid.net:587] SMTP connection ended  (state=4)
[DEBUG] [-] [core] [smtp_client_pool] 502FB3D1-C7D6-4938-9D76-64DFE8E2F0FD: [smtp.sendgrid.net:587] SMTP connection closed  (state=4)
[INFO] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] client [127.0.0.1] half closed connection
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] running reset_transaction hooks
[DEBUG] [9D97C92C-6955-48B1-9DFF-DDC7BC5C1161.1] [core] running disconnect hooks

**Comment by Dexus** [comment]
please provide your nodejs version, openssl version and system.

`echo "Haraka | $(haraka -v)"; echo " --- | :--- "; echo "Node | $(node -v)"; echo "OS | $(uname -a)"; echo "openssl | $(openssl version)"`

**Comment by ajay-psd** [comment]
Please find below version details

Haraka | Haraka.js — Version: 2.8.9
 --- | :---
Node | v6.9.1
OS | Linux host1 2.6.32-358.14.1.el6.x86_64 #1 SMP Mon Jun 17 15:54:20 EDT 2013 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.0.1e-fips 11 Feb 2013

**Comment by msimerson** [comment]
This **might** be related to coopernurse/node-pool#137.  I will bump the version of node-pool in Haraka 2.8.12 and see if this helps.

**Comment by msimerson** [comment]
This issue might be fixed or improved by #1803, since it's forwarding to TLS enabled hosts that require auth.

**Comment by msimerson** [comment]
Closing. Please try with Haraka 2.8.12 or newer. Reopen this issue of the problem persists.
